mocktracer/propagation.go

```diff
@@ -52,7 +55,11 @@ func (t *TextMapPropagator) Inject(spanContext MockSpanContext, carrier interfac
 	writer.Set(mockTextMapIdsPrefix+"sampled", fmt.Sprint(spanContext.Sampled))
 	// Baggage:
 	for baggageKey, baggageVal := range spanContext.Baggage {
-		writer.Set(mockTextMapBaggagePrefix+baggageKey, baggageVal)
+		safeVal := baggageVal
+		if t.HTTPHeaders {
+			safeVal = url.QueryEscape(baggageVal)
+		}
+		writer.Set(mockTextMapBaggagePrefix+baggageKey, safeVal)
```

Comment:
The keys are a problem, too. IMO, the only truly robust thing is to have the Tracer implementation completely control the keyspace. This either becomes `ot-baggage-item-0`, `ot-baggage-item-1`, etc, or we encode the entire lot as `ot-baggage-map` or similar. I prefer the latter, FWIW.